proto/minder/v1/minder.proto

```diff
@@ -920,6 +933,25 @@ message UpdateProfileResponse {
     Profile profile = 1;
 }
 
+message PatchProfileRequest {
+    // The context in which the patch is applied. Provided explicitly
+    // so that the patch itself can be minimal and contain only
+    // the attribute to set, e.g. remediate=true
+    Context context = 1;
```

Comment:
If the context is required, isn't it the same if the frontend passes in the nested context under profile?
Or is it not enforced if it's nested?

Reply:
I thought it would be confusing to include it in the body that represents what is going to be patched (changed), same as the ID.

btw @evankanderson pointed me to [https://grpc-ecosystem.github.io/grpc-gateway/docs/mapping/patch_feature/](the grpc PATCH) docs that say that the method is confusingly not supposed to be `patch` but `put`. I'll test how the method behaves then.

(I'll welcome another opinion on the separate context and ID in the meantime of course)

Reply:
I shouldn't reply on a Friday evening, sorry

> I thought it would be confusing to include it in the body that represents what is going to be patched (changed), same as the ID.
> 
> btw @evankanderson pointed me to [grpc patch docs](https://grpc-ecosystem.github.io/grpc-gateway/docs/mapping/patch_feature) that say that the method is confusingly not supposed to be `patch` but `put`. I'll test how the method behaves then.
>

I misread Evan's link, the method should be patch, what the example really had was:
```
rpc UpdateV2(UpdateV2Request) returns (google.protobuf.Empty) {
 option (google.api.http) = {
   put: "/v2/example/a_bit_of_everything/{abe.uuid}"
   body: "abe"
   additional_bindings {
     patch: "/v2/example/a_bit_of_everything/{abe.uuid}"
     body: "abe"
   }
 };
}
```

but both your comment and Evan's link make me think that just reusing the Update method and require an update field might be better. I'll try going down that route and push a new patch version.

Reply:
In the end I settled on `"/api/v1/profile/{id}` handling PATCH and accepting the context in a separate part of the call such as query parameters.

I still think the explicit context is somehow cleaner than the alternative which would be to pass both the context and the patch in the same message:
```
curl -XPATCH -H "Authorization: Bearer $MINDER_BEARER_TOKEN" 'http://localhost:8080/api/v1/profile' -d '{ "patch": {"context": { "provider": "github", "project": "a34dc21e-0164-4508-ada1-32419bc72d65" },"id": "9ad30fe9-d1ba-495c-9ca4-f99a97374d61", "remediate":"on" }}'
```
basically it seems like the case with the embedded context is useful for a git-ops flow like you said on Slack the other day, but I'm not sure if the patch method would be a part of such workflow, it seems really more like an imperative call. That said, I do have a local branch implementing that.